python/pyarrow/scalar.pxi

```diff
@@ -791,8 +791,11 @@ cdef class MapScalar(ListScalar):
         """
         Iterate over this element's values.
         """
+        return self
+
+    def __next__(self):
         arr = self.values
-        if array is None:
+        if arr is None:
             raise StopIteration
         for k, v in zip(arr.field('key'), arr.field('value')):
             yield (k.as_py(), v.as_py())
```

Comment:
Do you actually test these changes before pushing them?

You can't use `yield` in a `__next__` function.
Either you write `__iter__` as a generator (using `yield`), or you write a pair of `__iter__` and `__next__` functions (without `yield`). You can't mix the two styles.

Reply:
And FWIW, for this case, I would leave the `__iter__` as is (in the other cases such as RecordBatchReader, we already have a public "next" method that can just be called from `__next__`)

Reply:
Sorry. I didn't test on local before I pushed a commit because I don't have enough time yesterday.

I reverted the `__iter__` + `__next__` change.